src/tools/inner_node.rs

```diff
@@ -118,8 +122,24 @@ impl InnerNode {
 
     /// Connects to the target address.
     pub async fn connect(&self, target: SocketAddr) -> io::Result<()> {
-        self.node.connect(target).await?;
-        Ok(())
+        self.node.connect(target).await
+    }
+
+    /// Connects to the target address using outgoing_addr as source address/port.
+    /// Sets `reuse_addr` and `reuse_port` on the outgoing socket.
+    pub async fn connect_with_outgoing_addr(
```

Comment:
just maybe? Not sure
```suggestion
    pub async fn connect_from(
```